hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -180,6 +182,28 @@ public OmSnapshot load(@Nonnull String snapshotTableKey)
         // see if the snapshot exists
         snapshotInfo = getSnapshotInfo(snapshotTableKey);
 
+        // Block snapshot from loading when it is no longer active e.g. DELETED,
+        // unless this is called from SnapshotDeletingService.
+        // TODO: [SNAPSHOT] However, snapshotCache.get() from other requests
```

Comment:
If SDS can access this , then it can load it into the cache and other readers can access this via the  common cache.
One idea is to introduce a flag in OmSnapshot like isAccessedBySDS and set it to true while SDS tries to load it , if other readers try to access, we can quickly check this flag and restrict access.

Reply:
> One idea is to introduce a flag in OmSnapshot like isAccessedBySDS and set it to true while SDS tries to load it , if other readers try to access, we can quickly check this flag and restrict access.

@sadanand48 Actually this roughly becomes the same idea as the first revision of the patch, where I check snapshot status right before every single (read) operation. While here is just checking this new flag.

Reply:
Yeah right, like you commented we could also try with the separate cache for SDS access but I guess it can be done in another jira. Thanks @smengcl for explaining.